File: 1737-maximum-nesting-depth-of-the-parentheses/1737-maximum-nesting-depth-of-the-parentheses.cpp
class Solution {
public:
    int maxDepth(string s) {
        int ans=0,i=0,count=0;
       
        while(s[i] != '\0'){
          if(s[i] == '(')count++;
          else if(s[i] == ')' &&count>0){
            ans = max(ans,count); count--;
          }
          i++;
        }
        return ans;
    }
};

File: 1737-maximum-nesting-depth-of-the-parentheses/1737-maximum-nesting-depth-of-the-parentheses.java
class Solution {
    public int maxDepth(String s) {
        int cnt =0,maxcnt = 0;
        for(int i=0;i<s.length();i++){
        if(s.charAt(i) == '('){
            cnt++;
            maxcnt = Math.max(maxcnt,cnt);
        }
        else if(s.charAt(i) == ')') cnt--;
        if(cnt<0) cnt = 0;
        }
        return maxcnt;
    }
}